tp/src/state.rs

```diff
@@ -98,39 +99,43 @@ impl<'a> SabreState<'a> {
     ) -> Result<(), ApplyError> {
         let address = make_contract_address(name, version)?;
         let d = self.context.get_state_entry(&address)?;
-        let mut contract_list = match d {
-            Some(packed) => match protobuf::parse_from_bytes(packed.as_slice()) {
-                Ok(contracts) => contracts,
+        let mut contracts = match d {
+            Some(packed) => match ContractList::from_bytes(packed.as_slice()) {
+                Ok(contracts) => contracts.contracts().to_vec(),
```

Comment:
I don't think you need to convert this to a vec, since you're don't need ownership until you've filtered them.